frontend/src/components/Admin/Table/Table.js

```diff
@@ -0,0 +1,214 @@
+import { useTable, useFilters, useSortBy, usePagination } from "react-table";
+import { useState } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { 
+  faAnglesRight,
+  faAnglesLeft,
+  faArrowUpAZ,
+  faArrowDownZA,
+  faArrowRotateRight
+} from "@fortawesome/free-solid-svg-icons";
+import { useNavigate } from "react-router-dom"
+
+const Table = ({ columns, data }) => {
+  const right = <FontAwesomeIcon icon={faAnglesRight} />;
+  const left = <FontAwesomeIcon icon={faAnglesLeft} />;
+  const asc = <FontAwesomeIcon icon={faArrowUpAZ} />;
+  const desc = <FontAwesomeIcon icon={faArrowDownZA} />;
+  const reset = <FontAwesomeIcon icon={faArrowRotateRight} />;
+  const navigate = useNavigate()
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    page,
+    nextPage,
+    previousPage,
+    canPreviousPage,
+    canNextPage,
+    pageOptions,
+    setFilter,
+    state,
+    gotoPage,
+    pageCount,
+    setPageSize,
+    prepareRow,
+    setAllFilters,
+  } = useTable(
+    {
+      columns,
+      data,
+      initialState: { pageIndex: 0, pageSize: 25 },
+    },
+    useFilters,
+    useSortBy,
+    usePagination
+  );
+  const [searchInput, setSearchInput] = useState("");
+  const [filterInput, setFilterInput] = useState("id");
+  const { pageIndex, pageSize } = state;
+
+  const handleSearchChange = (e) => {
+    const value = e.target.value || "";
+    setFilter(filterInput, value);
+    setSearchInput(value);
+  };
+
+  const handleFilterChange = (e) => {
+    const value = e.target.value || "";
+    setFilterInput(value);
+  };
+
+  const getSearchChoices = () => {
+    if (!data || columns.length === 0) return [];
+
+    return columns[0].columns
+  };
+
+  const handleReset = () => {
+    setSearchInput("")
+    setAllFilters([])
+  }
+
+  const handleLink = (id) => {
+    navigate(id)
+  }
+
+  return (
+    <div className="m-2">
+      <div className="flex flex-row justify-between">
+        <div>
+          <input
+            value={searchInput}
+            onChange={handleSearchChange}
+            placeholder={"Search name"}
```

Comment:
Curly braces aren't necessary for the placeholder value unless you want to use a variable